src/views/Viewer.vue

```diff
@@ -736,6 +743,11 @@ export default {
 			this.Viewer.onClose()
 		},
 
+		onDownload() {
+			const { basename, filename } = this.currentFile
+			const url = OCA.Files.Files.getDownloadUrl(basename, dirname(filename))
+			location.href = url
+		},
```

Comment:
Files is not always available when viewer is here

You need to just link the davPath with an `ActionLink`.
I would move [the davPath method](https://github.com/nextcloud/viewer/blob/47a7d33dcf7f7445211265e7b68f17645c1fe10d/src/mixins/PreviewUrl.js#L47-L53) into [the File model](https://github.com/nextcloud/viewer/blob/f23e5234acc55928ac3fcc3b323aab6dae2662db/src/models/file.js)

Then I would be able to access it via `this.currentFile.davPath`

Reply:
Can we access to the File Model in the [PreviewUrl#previewPath](https://github.com/nextcloud/viewer/blob/47a7d33dcf7f7445211265e7b68f17645c1fe10d/src/mixins/PreviewUrl.js#L34-L39) method? If I am not wrong, it is used as a mixin in [Mime](https://github.com/nextcloud/viewer/blob/f23e5234acc55928ac3fcc3b323aab6dae2662db/src/mixins/Mime.js) which does not have any File model.

What rather about declaring a dav link generation function in `src/utils/fileUtils.js`? Or would it have benefits to declare it as a file model property or method?

Reply:
> What rather about declaring a dav link generation function in `src/utils/fileUtils.js`? Or would it have benefits to declare it as a file model property or method?

You could, but having it in the model would make it available automatically within every file.
I would put the method in fileUtil for code quality.
But call the davPath generation method from the file model.

Then `this.davPath` will be automatically available in PreviewUrl mixin as we bind the full File element to the component https://github.com/nextcloud/viewer/blob/04242f8aa69d04ae283b564342d4b8407dacc577/src/views/Viewer.vue#L86
Meaning any property within the object will be available from the components :)

Reply:
Thanks for your hint! I tried to take into account your remarks in 2983947

I couldn't manage to access to the davPath through `this.davPath` in PreviewUrl despite what you mentioned (I checked that thanks to a breakpoint).

AFAICT, `PreviewUrl` is used as a Mixin of `Mime`. If I am not wrong, the latter is used as a Mixin of the component passed to `registerHandler`.

Is it that component which takes the `currentFile` binding?

Reply:
Up please :)

Reply:
Sorry, I missed that comment.
I don't understand it.

Reply:
I added some reviews.